src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,258 @@
+package duke;
+
+import duke.Tasks.BaseTask;
+
+
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+/**
+ * This class contains the main Duke class that runs the chat bot.
+ */
+public class Duke {
+
+    /** Tracks whether the exit command was used. */
+    private boolean isExited;
+
+
+    /** Storage and Management of Duke Objects. */
+    private static Duke dukeInstance;
+
+    private DukeCommandParser currDukeCmdParser;
+    private DukeStorageManager currStorageMgr;
+    private DukeUi currUiCtrl;
+    private DukeListMgr currDukeList;
+
+    private Duke() {
+        this.currDukeList = new DukeListMgr();
+        this.currUiCtrl = new DukeUi();
+        this.currDukeCmdParser = new DukeCommandParser();
+        this.isExited = false;
+
+        // Processes the path to use to access storage file
+        try {
+            Path xmlPath = Paths.get("dukeDocs", "listSave1.xml");
+            //System.out.println(xmlPath.getFileName());
+            this.currStorageMgr = new DukeStorageManager(xmlPath);
+        } catch (InvalidPathException e) {
+            // If there was an error loading the save file, then create a new empty one.
+            this.currStorageMgr = new DukeStorageManager();
+        } catch (DukeExceptionBase e) {
+            // There is no Duke instance to print the error msg yet
+            System.out.println(e);
+            // If there was an error loading the save file, then create a new empty one.
+            this.currStorageMgr = new DukeStorageManager();
+        }
+
+
+    }
+
+    /**
+     * Use to get the current instance of Duke that is running.
+     * Creates a new instance of Duke if there is none running.
+     * @return the Duke object that is running.
+     */
+    public static Duke getCurrDuke() {
+        if (dukeInstance == null) {
+            dukeInstance = new Duke();
+        }
+        return dukeInstance;
+    }
+
+    /**
+     * Getter for Duke's Command Parser.
```

Comment:
Should the first word be a verb like 'gets'? I noticed the same issue in several other places too